holidays/countries/afghanistan.py

```diff
@@ -0,0 +1,99 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars.julian import JULIAN_CALENDAR
+from holidays.groups import (
+    ChristianHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+    StaticHolidays,
+)
+from holidays.observed_holiday_base import ObservedHolidayBase
+
+
+class Afghanistan(
+    ObservedHolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolidays, StaticHolidays
+):
+    """
+    https://en.wikipedia.org/wiki/Public_holidays_in_Afghanistan
+    https://www.timeanddate.com/holidays/afghanistan/
+    """
+
+    country = "AF"
+    default_language = "fa"
+    supported_languages = ("fa", "ps_AF")
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self, JULIAN_CALENDAR)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(self)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        if self._year >= 1989:
+            self._add_holiday_feb_15(tr("روز آزادی"))  # Liberation Day
+
+        self._add_holiday_mar_21(tr("نوروز"))  # Nowruz
+
+        if 1992 <= self._year <= 1996:
+            self._add_holiday_apr_28(tr("روز شکست مجاهدین"))  # Defeat of Mujahideen Day
+
+        if self._year >= 1889:
+            self._add_labor_day(tr("روز جهانی کارگر"))  # International Workers' Day
+
+        if 1989 <= self._year <= 2021:
+            self._add_holiday_aug_15(tr("روز پیروزی شوروی بر افغانستان"))  # Soviet Victory Day
+
+        if self._year >= 1919:
```

Comment:
The 1973 Afghan coup d'état, also called by Afghans as the Coup of 26 Saratan [[4]](https://en.wikipedia.org/wiki/1973_Afghan_coup_d'%C3%A9tat#cite_note-4) and self-proclaimed as the Revolution of 26 Saratan 1352,[[a]](https://en.wikipedia.org/wiki/1973_Afghan_coup_d'%C3%A9tat#cite_note-5)[[5]](https://en.wikipedia.org/wiki/1973_Afghan_coup_d'%C3%A9tat#cite_note-6) was led by [Army](https://en.wikipedia.org/wiki/Afghan_Army) [General](https://en.wikipedia.org/wiki/General_officer) and prince [Mohammad Daoud Khan](https://en.wikipedia.org/wiki/Mohammad_Daoud_Khan) against his cousin, King [Mohammad Zahir Shah](https://en.wikipedia.org/wiki/Mohammad_Zahir_Shah), on 17 July 1973, which resulted in the establishment of the [Republic of Afghanistan](https://en.wikipedia.org/wiki/Republic_of_Afghanistan_(1973%E2%80%931978)) under a one-party system led by Daoud Khan.[[6]](https://en.wikipedia.org/wiki/1973_Afghan_coup_d'%C3%A9tat#cite_note-NYT-7)[[3]](https://en.wikipedia.org/wiki/1973_Afghan_coup_d'%C3%A9tat#cite_note-auto1-3)

This party became the sole focus of political activity in the country. In 1974, the new flag of [Afghanistan](https://en.wikipedia.org/wiki/Afghanistan) was introduced, as well its emblem, with a flag raising ceremony being held outside of the [Arg Presidential Palace](https://en.wikipedia.org/wiki/Arg_Presidential_Palace).[[7]](https://en.wikipedia.org/wiki/Republic_of_Afghanistan_(1973%E2%80%931978)#cite_note-7) In January 1977, a [loya jirga](https://en.wikipedia.org/wiki/Loya_jirga) was convened following the [Constitutional Assembly election](https://en.wikipedia.org/wiki/1977_Afghan_Constitutional_Assembly_election), and approved a new constitution establishing a presidential [one-party state](https://en.wikipedia.org/wiki/One-party_state), with political opposition being suppressed, sometimes violently.[[3]](https://en.wikipedia.org/wiki/Republic_of_Afghanistan_(1973%E2%80%931978)#cite_note-daoud-3)

I don't have any quotation stating that this event affected Afghan Independence Day.

Reply:
[Here](https://www.officeholidays.com/holidays/afghanistan/afghanistan-independence-day) it is said that

> In 1973, a bloodless coup d’état ended the Afghan monarchy, and the date of jeshn moved to the anniversary of the coup on 26 Saratan, 1352 (July 17th 1973).

However, [here](https://www.afghanistan-analysts.org/en/reports/context-culture/symbolism-of-a-day-a-century-of-changing-independence-day-celebrations-in-afghanistan/) is evidences that Independence Day was celebrated as before:

> I remember jashn-e esteqlal towards the end of Daud time, especially in the years 1355 [1976] and 1356 [1977]. The jashn continued for seven nights and days, especially after evening around 7 or 8 pm till the following morning, because 28 Asad [19 August], roz-e esteqlal [independence day], falls in the summer.

So let's leave it as it is.